uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/RepositoryServiceImpl.java

```diff
@@ -389,43 +407,52 @@ public void removeRepositories(final Space space,
                                    repo -> {
-                                   });
+                                   },
+                                   false);
             }
         } catch (final Exception e) {
             logger.error("Error while removing repositories",
                          e);
             throw new RuntimeException(e);
         } finally {
-            configurationService.endBatch();
+            configStorage.endBatch();
         }
     }
 
     private void doRemoveRepository(final OrganizationalUnit orgUnit,
                                     final String alias,
-                                    final ConfigGroup thisRepositoryConfig,
-                                    final Consumer<Repository> notification) throws Exception {
-        if (thisRepositoryConfig != null) {
-            configurationService.removeConfiguration(thisRepositoryConfig);
-        }
+                                    final Optional<org.guvnor.structure.organizationalunit.config.RepositoryInfo> thisRepositoryConfig,
+                                    final Consumer<Repository> notification,
+                                    final boolean lock) {
 
-        final Repository repo = configuredRepositories.remove(orgUnit.getSpace(),
-                                                              alias);
-        if (repo != null) {
-            notification.accept(repo);
+        SpaceConfigStorage configStorage = this.spaceConfigStorage.get(orgUnit.getName());
 
-            Branch defaultBranch = repo.getDefaultBranch().orElseThrow(() -> new IllegalStateException("Repository should have at least one branch."));
-            ioService.delete(convert(defaultBranch.getPath()).getFileSystem().getPath(null));
-        }
+        try {
+            if (lock) {
+                configStorage.startBatch();
+            }
+
+            Repository repo = this.configuredRepositories.getRepositoryByRepositoryAlias(orgUnit.getSpace(),
+                                                                                         alias);
+
+            if (repo != null) {
+                notification.accept(repo);
+            }
 
-        //Remove reference to Repository from Organizational Units
-        for (Repository repository : orgUnit.getRepositories()) {
-            if (repository.getAlias().equals(alias)) {
-                organizationalUnitService.removeRepository(orgUnit,
-                                                           repository);
-                metadataStore.delete(alias);
+            //Remove reference to Repository from Organizational Units
+            for (Repository repository : orgUnit.getRepositories()) {
+                if (repository.getAlias().equals(alias)) {
+                    organizationalUnitService.removeRepository(orgUnit,
+                                                               repository);
+                    metadataStore.delete(alias);
```

Comment:
@adrielparedes here the call for metadatastore ^

Reply:
I keep that in there, do you think I should move this to DeleteWorker?